### PR TITLE
template:generic-template-messages

### PR DESCRIPTION
**Pull request by AmishaBisht**
## Summary
 Target issue is #2783
added a generic function for sending template via expression into the template file.

## Checklist
  Before submitting a pull request, please ensure that you mark these task.

- [ ] Ran `mix setup` in the repository root and test.
- [ ] If you've fixed a bug or added code that is tested and has test cases. 
- [ ] In the case of adding a new API, you added a **postman** request for that.
- [ ] Coding standard and conventions are followed. https://docs.google.com/document/d/1rfU33IjS-ioiIH0TBWpxoLsdyyYdI1tDrqr5HCRIb98/edit?usp=sharing

## Notes
 Please add here if any other information is required for the reviewer.

### CODE REVIEW

File: lib/glific/templates.ex [hunk 1]
```diff
@@ -694,4 +694,18 @@ defmodule Glific.Templates do
       error -> {:ok, %{message: error}}
     end
   end
+
+  @doc """
+  get template form EEx based on variables
+  """
+  @spec template(integer(), String.t() | nil) :: binary
+  def template(template_uuid, variables \\ nil) do
```
Comment (mdshamoon) [inline lib/glific/templates.ex]: variables will be a list of strings

**Review by mdshamoon — COMMENTED** [review]
Also can you test this in a flow if this is working fine.

File: test/glific/templates_test.exs [hunk 2]
```diff
@@ -1327,5 +1327,28 @@ defmodule Glific.TemplatesTest do
...
+    test "template EEx form with variables should be JSON encoded" do
+      result = %{
+        uuid: "uuid",
+        name: "Template",
+        variables: ["variable"],
+        expression: nil
+      }
+
+      assert Templates.template("uuid", ["variable"]) == Jason.encode!(result)
+      IO.inspect(assert Templates.template("uuid", ["variable"]) == Jason.encode!(result))
```
Comment (AkhileshNegi) [inline test/glific/templates_test.exs]: remove inspect

File: lib/glific/clients/arogya_world.ex [hunk 3]
```diff
@@ -302,10 +302,10 @@ defmodule Glific.Clients.ArogyaWorld do
   end
 
   @doc """
-  get template form EEx based on variables
+  get template form EEx based on  variables
```
Comment (mdshamoon) [inline lib/glific/clients/arogya_world.ex]:
```suggestion
  get template from EEx based on  variables
```

File: lib/glific/clients/arogya_world.ex [hunk 4]
```diff
@@ -302,10 +302,10 @@ defmodule Glific.Clients.ArogyaWorld do
...
   """
-  @spec template(integer(), String.t()) :: binary
-  def template(template_uuid, variables) do
+  @spec template(integer(), [any()]) :: binary
```
Comment (mdshamoon) [inline lib/glific/clients/arogya_world.ex]:
```suggestion
  @spec template(integer(), [String.t()]) :: binary
```

File: lib/glific/templates.ex [hunk 5]
```diff
@@ -694,4 +694,18 @@ defmodule Glific.Templates do
       error -> {:ok, %{message: error}}
     end
   end
+
+  @doc """
+  get template form EEx based on  variables
```
Comment (mdshamoon) [inline lib/glific/templates.ex]:
```suggestion
  get template from EEx based on  variables
```

File: lib/glific/templates.ex [hunk 6]
```diff
@@ -694,4 +694,18 @@ defmodule Glific.Templates do
...
+  """
+  @spec template(integer(), [any()]) :: binary
```
Comment (mdshamoon) [inline lib/glific/templates.ex]:
```suggestion
  @spec template(integer(), [String.t()]) :: binary
```

**Review by mdshamoon — CHANGES_REQUESTED** [review]

**Review by AkhileshNegi — APPROVED** [review]

**Pull request merged**